### DISCUSSION TITLE
FCM iOS: Where is the "from" property in the payload?

### DISCUSSION THREAD

**Discussion by pjivers-bom** [invertase/react-native-firebase#5292] [Q&A]
I have a background message handler set up like so:

```ts
const onMessageReceived = async (message: FirebaseMessagingTypes.RemoteMessage) => {
  if (message.from && message.from.startsWith('/topics/warning')) {
    onWarningMessageReceived(message)
  } else {
    onGenericMessageReceived(message)
  }
}

messaging().setBackgroundMessageHandler(onMessageReceived)
```

As you can see, it uses the "from" property included in the message to check the topic the message was sent from, and then delegates to a specific handler. This works great on Android, but for some reason the "from" property isn't present when the message is received on iOS.

I can see from the [types definition](https://github.com/invertase/react-native-firebase/blob/master/packages/messaging/lib/index.d.ts#L90) that "from" is optional, but it doesn't explain what circumstances it gets attached to the message. There are other properties such as "contentAvailable" that are iOS-only, but "from" isn't described in that way.

Is there a reason I get this property in a message on Android but not on iOS? I've noticed that "sentTime" and "ttl" are also missing when messages are received on iOS.

**Comment by mikehardy** [comment]
It appears we would pass it through if it came in:

https://github.com/invertase/react-native-firebase/blob/3b482401e90fa573279ce8066fef9bdbcd6f59bd/packages/messaging/ios/RNFBMessaging/RNFBMessagingSerializer.m#L64-L68

I don't have any solid answers but I'd rummage around in firebase-ios-sdk looking for this:

https://github.com/firebase/firebase-ios-sdk/blob/8b3ee5f12903185411c6cb839341fb34c31e4daa/FirebaseMessaging/Sources/FIRMessagingConstants.m#L19
